# Descriptive Statistics - Produksi Padi di Sumatera
Mengidentifikasi hubungan antar fitur pada data Produksi Padi di Sumatera dengan menerapkan salah satu konsep descriptive statistics, yaitu Data Relationship.

Dataset source : https://www.kaggle.com/datasets/ardikasatria/datasettanamanpadisumatera

Dataset memiliki 7 kolom, yaitu:
- Provinsi
- Tahun
- Produksi
- Luas Panen
- Curah Hujan
- Kelembapan
- Suhu rata-rata



## Import Packages

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt

## Read CSV File

In [2]:
data = pd.read_csv('https://raw.githubusercontent.com/syifagit/Belajar-Analisis-Data-Dengan-Python/main/Dataset/Dataset%20Tanaman%20Padi%20Sumatera%2C%20Indonesia.csv')
data = pd.DataFrame(data)

In [3]:
data.head()

Provinsi  Tahun   Produksi  Luas Panen  Curah hujan  Kelembapan  \
0     Aceh   1993  1329536.0    323589.0       1627.0       82.00   
1     Aceh   1994  1299699.0    329041.0       1521.0       82.12   
2     Aceh   1995  1382905.0    339253.0       1476.0       82.72   
3     Aceh   1996  1419128.0    348223.0       1557.0       83.00   
4     Aceh   1997  1368074.0    337561.0       1339.0       82.46   

   Suhu rata-rata  
0           26.06  
1           26.92  
2           26.27  
3           26.08  
4           26.31

## Data Relationship

### Correlation
- *Correlation* mengidentifikasi hubungan 2 fitur numerik dalam sebuah data.
- *Correlation* memiliki nilai dengan rentang -1 hingga 1.
- Memiliki 3 jenis, yaitu:
  - *Positive correlation* : merepresentasikan korelasi bersesuaian.
  - *Negative correlation* : merepresentasikan korelasi berlawanan.
  - *No correlation* : merepresentasikan tidak adanya korelasi.


In [4]:
data.corr()

<ipython-input-4-c44ded798807>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  data.corr()


Tahun  Produksi  Luas Panen  Curah hujan  Kelembapan  \
Tahun           1.000000  0.182527   -0.045951    -0.047645   -0.033474   
Produksi        0.182527  1.000000    0.905622    -0.042129   -0.052316   
Luas Panen     -0.045951  0.905622    1.000000    -0.092975   -0.061121   
Curah hujan    -0.047645 -0.042129   -0.092975     1.000000    0.056466   
Kelembapan     -0.033474 -0.052316   -0.061121     0.056466    1.000000   
Suhu rata-rata  0.004923  0.041160    0.115726    -0.228699   -0.407799   

                Suhu rata-rata  
Tahun                 0.004923  
Produksi              0.041160  
Luas Panen            0.115726  
Curah hujan          -0.228699  
Kelembapan           -0.407799  
Suhu rata-rata        1.000000

Dari tabel di atas, terlihat bahwa:
- Fitur Produksi dan Luas Panen memiliki nilai korelasi terbesar yaitu 0.905622	-> **Positive Correlation**, yang berarti bahwa Fitur Produksi memiliki korelasi yang bersesuaian dengan Fitur Luas Panen. Semakin besar luas panen, makin semakin besar pula hasil produksinya.
- Fitur Curah Hujan, Kelembapan, dan Suhu rata-rata termasuk ke dalam **No Correlation** jika dihubungkan dengan Fitur Produksi karena nilai korelasinya berada pada rentang -0.05 hingga 0.04.

### Covariance
- Selain menggunakan correlation, kita juga dapat menggunakan *Covariance* untuk mengidentifikasi hubungan antara 2 fitur dalam sebuah data.
- Memiliki 3 jenis, yaitu:
  - *Positive Covariance* : merepresentasikan hubungan bersesuaian.
  - *Negative Covariance* : merepresentasikan hubungan berlawanan.
  - *Zero Covariance* : merepresentasikan tidak adanya hubungan satu sama lain.

In [5]:
data.cov()

<ipython-input-5-72e63cb34c7c>:1: FutureWarning: The default value of numeric_only in DataFrame.cov is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  data.cov()


Tahun      Produksi    Luas Panen   Curah hujan  \
Tahun           6.554260e+01  1.716192e+06 -8.658591e+04 -3.980610e+02   
Produksi        1.716192e+06  1.348821e+12  2.448026e+11 -5.049197e+07   
Luas Panen     -8.658591e+04  2.448026e+11  5.417310e+10 -2.233204e+07   
Curah hujan    -3.980610e+02 -5.049197e+07 -2.233204e+07  1.064967e+06   
Kelembapan     -1.322130e+00 -2.964254e+05 -6.940424e+04  2.842872e+02   
Suhu rata-rata  4.771300e-02  5.722136e+04  3.224281e+04 -2.825147e+02   

                   Kelembapan  Suhu rata-rata  
Tahun               -1.322130        0.047713  
Produksi       -296425.416419    57221.362528  
Luas Panen      -69404.236608    32242.812252  
Curah hujan        284.287243     -282.514680  
Kelembapan          23.801517       -2.381540  
Suhu rata-rata      -2.381540        1.432907